### PR TITLE
doc: Improve the documentation of "tracked users"

### PR DESCRIPTION
**Pull request by richvdh**
I had no idea what a tracked user was, so I've attempted to clarify this
somewhat for future readers.

### CODE REVIEW

File: bindings/matrix-sdk-crypto-js/src/machine.rs [hunk 1]
```diff
@@ -175,16 +178,23 @@ impl OlmMachine {
...
+    /// The OlmMachine maintains a list of users whose devices we are keeping
+    /// track of: these are known as "tracked users". These must be users
+    /// that we share a room with, so that the server sends us updates for
+    /// their device lists.
+    ///
+    /// # Arguments
     ///
-    /// `users` is an iterator over user IDs that should be marked for
-    /// tracking.
+    /// * `users` - An iterator over user ids that should be added to the list
```
Comment (poljar) [inline bindings/matrix-sdk-crypto-js/src/machine.rs:190]:
```suggestion
    /// * `users` - An array of user ids that should be added to the list
```

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 2]
```diff
@@ -295,14 +295,17 @@ impl OlmMachine {
...
+    /// See [`update_tracked_users`](#method.update_tracked_users) for more
+    /// information.
     pub fn tracked_users(&self) -> HashSet<OwnedUserId> {
         self.store.tracked_users()
     }
 
     /// Get the outgoing requests that need to be sent out.
     ///
-    /// This returns a list of `OutGoingRequest`, those requests need to be sent
+    /// This returns a list of `OutgoingRequest`. Those requests need to be sent
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/machine.rs:308]: Let's link to the type as well: 

```suggestion
    /// This returns a list of [`OutgoingRequest`]. Those requests need to be sent
```

**Review by poljar — CHANGES_REQUESTED** [review]
I wonder if we should, instead of copying the docs around, ignore the fact that the arguments aren't exactly the same type, and put the docs into a markdown file which all crates then include.

Something along the lines of the docs for the `encryption` module in the main crate, [this](https://github.com/matrix-org/matrix-rust-sdk/blob/1e2fd02478e39cd7183041803633fbf1e1962fcd/crates/matrix-sdk/src/docs/encryption.md) file gets included in the docs here:

https://github.com/matrix-org/matrix-rust-sdk/blob/1e2fd02478e39cd7183041803633fbf1e1962fcd/crates/matrix-sdk/src/encryption/mod.rs#L16

@jplatte are you aware of any problems of multiple crates sharing the same doc fragments? 

I think there might be problems when we publish the crate, that the fragments need to be part of the source package. But this is about the bindings sharing the docs of the crypto crate, and we only publish the crypto crate.

**Comment by jplatte** [comment]
> @jplatte are you aware of any problems of multiple crates sharing the same doc fragments?
>
> I think there might be problems when we publish the crate, that the fragments need to be part of the source package. But this is about the bindings sharing the docs of the crypto crate, and we only publish the crypto crate.

Yeah, publishing is where this could get problematic. Cargo modifies `Cargo.toml` to not have any workspace references when publishing, but I'm pretty sure it doesn't account for `include_str!` / `include!`s that point outside of the directory that holds `Cargo.toml`.

If we never want to publish the bindings to a registry (not sure why we would), it should be fine to do this with the doc file living inside `crates/matrix-sdk-crypto` somewhere.

**Comment by poljar** [comment]
> If we never want to publish the bindings to a registry (not sure why we would), it should be fine to do this with the doc file living inside `crates/matrix-sdk-crypto` somewhere.

Right, that's what I was thinking as well, thanks for confirming. The only thing that would suck in this case is that we don't see the docs anymore while editing the source.

**Comment by richvdh** [comment]
> I wonder if we should, instead of copying the docs around, ignore the fact that the arguments aren't exactly the same type, and put the docs into a markdown file which all crates then include.

yeah, it did seem silly to repeat the documentation in three places. To be honest I wondered if it would be adequate for the docs for the bindings to be very skeletal, with a link to https://docs.rs/matrix-sdk-crypto/0.6.0/matrix_sdk_crypto/struct.OlmMachine.html (or a more recent version). "See [OlmMachine docs] for more information".

**Comment by poljar** [comment]
> > I wonder if we should, instead of copying the docs around, ignore the fact that the arguments aren't exactly the same type, and put the docs into a markdown file which all crates then include.
> 
> yeah, it did seem silly to repeat the documentation in three places. To be honest I wondered if it would be adequate for the docs for the bindings to be very skeletal, with a link to https://docs.rs/matrix-sdk-crypto/0.6.0/matrix_sdk_crypto/struct.OlmMachine.html (or a more recent version). "See [OlmMachine docs] for more information".

Do the docs end up in the JS code or do you have to fish for them in the Rust sources, I know that [pyo3](https://pyo3.rs/) takes the Rust docstrings and converts them to Python docstrings. But I'm not sure that wasm-bindgen supports this.

In any case, I think that `uniffi` will support this eventually, so it would be nice if users of the bindings don't need to fish around the Rust docs. Thus the idea of the markdown fragments, that get included, seems to be the better path forward.

**Review by poljar — APPROVED** [review]
I'll open an issue about the doc fragments. Thanks, looks good. I took care of the remaining `cargo fmt` issue.

**Pull request merged**